partiql-lib/partiql-sprout/src/main/kotlin/org/partiql/sprout/generator/target/kotlin/spec/KotlinNodeSpec.kt

```diff
@@ -62,33 +70,39 @@ sealed class KotlinNodeSpec(
     class Product(
         val product: TypeDef.Product,
         val props: List<Prop>,
-        override val children: List<KotlinNodeSpec>,
+        override val nodes: List<KotlinNodeSpec>,
         clazz: ClassName,
-        types: List<TypeSpec> = emptyList(),
+        ext: MutableList<TypeSpec> = mutableListOf(),
     ) : KotlinNodeSpec(
         def = product,
         clazz = clazz,
         builder = TypeSpec.classBuilder(clazz),
         constructor = FunSpec.constructorBuilder(),
         companion = TypeSpec.companionObjectBuilder(),
-        types = types.toMutableList()
-    )
+        ext = ext,
+    ) {
+        override val children: List<KotlinNodeSpec> = nodes
```

Comment:
Do we not care about `Prop` here?

Reply:
Prop is separated at this point. Nested definitions (including inlines of prop) are assigend to no. Prop doesn't hold a typespec builder. Prop is simply a name-to-type holder.